admin/services/pupil-history-service/data-access/pupil-history.data.service.ts

```diff
@@ -0,0 +1,236 @@
+import moment from 'moment';
+const sqlService = require('../../data-access/sql.service')
+const uuidValidate = require('uuid-validate')
+const R = require('ramda')
+
+export interface ICheckData {
+  id: number,
+  createdAt: moment.Moment
+  updatedAt: moment.Moment,
+  pupilId: number,
+  checkCode: string,
+  checkWindowId: number,
+  checkFormId: number,
+  pupilLoginDate: null | moment.Moment,
+  receivedByServerAt: null | moment.Moment,
+  isLiveCheck: boolean,
+  received: boolean,
+  complete: boolean,
+  completedAt: null | moment.Moment,
+  processingFailed: boolean,
+  createdByUserId: number,
+  inputAssistantAddedRetrospectively: boolean,
+  resultsSynchronised: boolean
+}
+
+export interface IPupilData {
+  id: number,
+  createdAt: moment.Moment,
+  updatedAt: moment.Moment,
+  schoolId: number,
+  foreName: string,
+  middleNames: string,
+  lastName: string,
+  gender: string,
+  dateOfBirth: moment.Moment,
+  upn: string,
+  urlSlug: string,
+  groupId: number | null,
+  currentCheckId: number | null,
+  checkComplete: boolean,
+  restartAvailable: boolean,
+  attendanceId: number | null,
+  foreNameAlias: string | null,
+  lastNameAlias: string | null
+}
+
+export interface ISchoolData {
+  id: number,
+  createdAt: moment.Moment,
+  updatedAt: moment.Moment,
+  leaCode: number,
+  estabCode: number,
+  name: string,
+  pin: string | null,
+  pinExpiresAt: moment.Moment | null,
+  urlSlug: string,
+  urn: number,
+  dfeNumber: number
+}
+
+export interface IRestartData {
+  id: number,
+  createdAt: moment.Moment,
+  updatedAt: moment.Moment,
+  pupilId: number,
+  restartReason: string,
+  restartReasonCode: string,
+  checkId: number,
+  originCheckId: number
+}
+
+export interface IPupilHistoryData {
+  pupil: IPupilData,
+  checks: ICheckData[]
+  school: ISchoolData
+  restarts: IRestartData[]
+}
+
+export class PupilHistoryDataService {
+  public static async getPupil(pupilUuid: string): Promise<IPupilData> {
```

Comment:
why static? is this so they can be consumed easily in the existing javascript? or just a preference/design choice

In `tslib` we have interfaces for data services, and mocks complying to those interfaces are injected, which is nice and clean.  are you proposing a different way for `admin`?